hugr-core/src/std_extensions/arithmetic/int_types.rs

```diff
@@ -276,4 +276,29 @@ mod test {
         ConstInt::new_s(50, -2).unwrap_err();
         ConstInt::new_u(50, 2).unwrap_err();
     }
+
+    mod proptest {
+        use super::{ConstInt, LOG_WIDTH_MAX};
+        use ::proptest::prelude::*;
+        impl Arbitrary for ConstInt {
+            type Parameters = ();
+            type Strategy = BoxedStrategy<Self>;
+            fn arbitrary_with(_args: Self::Parameters) -> Self::Strategy {
+                let signed_strat = (..=LOG_WIDTH_MAX).prop_flat_map(|log_width| {
+                    use i64;
+                    let max_val = (2u64.pow(log_width as u32) / 2) as i64;
+                    let min_val = -max_val - 1;
```

Comment:
Hang on. So let's take `log_width == 5`. Then `max_val = 2u64.pow(5)/2` i.e. `16`. `min_val` is therefore `-17`. You're then creating a value in range `-17..16`...

I'd not expect `Self::new_s` to be happy if you generated `-17` or `16` though. But maybe you're not hitting this? I tried copying your `max_val` and `min_val` computation into a separate test and that lead me tothink my interpretation was correct...

(So, consider: rename `max_val` to `max_valp1`; then `let min_val = -max_valp1`; then use `min_val..<max_valp1`)